src/Pages/7 Writing Plugins/Examples.md

```diff
@@ -0,0 +1,371 @@
+    {
+        // Collect all tags from all pages
+        var allTags = context.GeneratedPages
+            .SelectMany(p => p.Page.Metadata.GetValueOrDefault("tags", "")
+                .Split(',', StringSplitOptions.RemoveEmptyEntries))
+            .GroupBy(tag => tag.Trim())
+            .ToDictionary(g => g.Key, g => g.Count());
+
+        // Generate tag cloud data file
+        var tagCloudPath = Path.Combine(context.OutputPath, "tag-cloud.json");
+        await File.WriteAllTextAsync(tagCloudPath, JsonSerializer.Serialize(allTags));
+        
+        logger?.LogInformation($"Generated tag cloud with {allTags.Count} unique tags");
+    }
+}
+```
+
+#### Content Processing Plugins
+
+**Code Highlighter Plugin:**
+```csharp
+public class CodeHighlighterPlugin : IBlakePlugin
+{
+    public async Task BeforeBakeAsync(BlakeContext context, ILogger? logger = null)
+    {
+        // Add syntax highlighting extension to Markdig
+        context.MarkdigPipelineBuilder.Extensions.Add(new SyntaxHighlightingExtension());
+    }
+}
+
+public class SyntaxHighlightingExtension : IMarkdownExtension
+{
+    public void Setup(MarkdownPipelineBuilder pipeline)
+    {
+        // Configure syntax highlighting
+        pipeline.Extensions.Add(new Markdig.Extensions.SyntaxHighlighting.SyntaxHighlightingExtension());
+    }
+}
+```
+
+**Table of Contents Plugin:**
+```csharp
+public class TocPlugin : IBlakePlugin
+{
+    public async Task BeforeBakeAsync(BlakeContext context, ILogger? logger = null)
+    {
+        foreach (var page in context.Pages)
+        {
+            var toc = GenerateTableOfContents(page.Content);
+            page.Metadata["tableOfContents"] = JsonSerializer.Serialize(toc);
+        }
+    }
+
+    private List<TocItem> GenerateTableOfContents(string content)
+    {
+        var toc = new List<TocItem>();
+        var lines = content.Split('\n');
+        
+        foreach (var line in lines)
+        {
+            if (line.StartsWith('#'))
+            {
+                var level = line.TakeWhile(c => c == '#').Count();
+                var title = line.Substring(level).Trim();
+                var anchor = GenerateAnchor(title);
+                
+                toc.Add(new TocItem { Level = level, Title = title, Anchor = anchor });
+            }
+        }
+        
+        return toc;
+    }
+}
+```
+
+### Learning Path Examples
+
+#### Beginner: Simple Metadata Plugin
+
+Start with a plugin that adds simple metadata:
+
+```csharp
+public class WordCountPlugin : IBlakePlugin
+{
+    public async Task BeforeBakeAsync(BlakeContext context, ILogger? logger = null)
+    {
+        foreach (var page in context.Pages)
+        {
+            var wordCount = page.Content
+                .Split(new[] { ' ', '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries)
+                .Length;
+                
+            page.Metadata["wordCount"] = wordCount.ToString();
```

Comment:
A few problems here. There is no property called `Pages` on the Blake context. There is `MarkdownPages`, and `GeneratedPages`.

`MarkdownPages` is populated when the context is instantiated and is available in the before bake hook. But the meta data dictionary is not included; that lives in the `GeneratedPages`, and is created during bake. So is not available in this hook, only in after bake.

Reply:
Note: please ensure all code examples are accurate, review this across all of them.